lib/flutter_typeahead.dart

```diff
@@ -628,7 +642,7 @@ class _TypeAheadFieldState<T> extends State<TypeAheadField<T>> {
 
     this._suggestionsBoxController = _SuggestionsBoxController(context);
 
-    (() async {
+    WidgetsBinding.instance.addPostFrameCallback((duration) async {
```

Comment:
That code just adds a callback to be executed after the frame. Originally the code used

```
     while (renderBox == null) {	
       await Future.delayed(Duration(milliseconds: 10));	
 
        renderBox = context.findRenderObject();	
     }	
 
      while (!renderBox.hasSize) {	
       await Future.delayed(Duration(milliseconds: 10));	
     }
```

By waiting until after the frame, renderBox will be ready and we do not need to loop and wait.

That addPostFrameCallback line and removing the above code does not have an impact on the suggestions list sizing and scrolling.

I added it in because it made the code cleaner. It doesn't have an impact on performance either way. The callback is only called once as well, so it works well for its purpose.

Reply:
Ohh interesting! thank you!